nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -174,6 +177,18 @@ ProtectedNiFiProperties readProtectedPropertiesFromDisk(File file) {
             rawProperties.load(inStream);
             logger.info("Loaded {} properties from {}", rawProperties.size(), file.getAbsolutePath());
 
+            // Trim whitespace from each property. If property is multi-line, remove anything after the first line break.
+            Set<String> keys = rawProperties.stringPropertyNames();
+            Iterator<String> itr = keys.iterator();
+            while(itr.hasNext()){
+                String key = itr.next();
+                String prop = rawProperties.getProperty(key);
+                if(!prop.isEmpty()){
+                    prop = Pattern.compile("\\s+$").matcher(prop).replaceFirst("");
```

Comment:
Probably better to define the Pattern once as a class-level static final field *or* use the String class' `replace` such as `prop.replace("\\s+$", "")`, although I'd guess using the pre-compiled Pattern would be slightly more efficient... but this code is probably only run once at startup anyway so it's unlikely a big deal (but I'd still prefer Pattern not to be compiled every time within the `while` loop)

Reply:
I've moved the pattern compilation out of the loop.